jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java

```diff
@@ -439,25 +459,27 @@ public void release(Connection connection)
 
     public boolean remove(Connection connection)
     {
-        return connectionPool.remove(connection);
-    }
-
-    public void close(Connection connection)
-    {
-        boolean removed = remove(connection);
+        boolean removed = connectionPool.remove(connection);
 
         if (getHttpExchanges().isEmpty())
         {
             tryRemoveIdleDestination();
         }
         else
         {
-            // We need to execute queued requests even if this connection failed.
-            // We may create a connection that is not needed, but it will eventually
-            // idle timeout, so no worries.
+            // We need to execute queued requests
+            // even if this connection was removed.
+            // We may create a connection that is not
+            // needed, but it will eventually idle timeout.
             if (removed)
-                process();
+                process(true);
```

Comment:
should this check that the queue is not 0?  Why acquire a connection if there are no waiting exchanges?

Reply:
The check for the queue is done few lines above. Do you want to do it again here?

Reply:
Opps didn't see that.  

An `else if (removed)` would be clearer that there is a condition above.

Reply:
I think an `else if` would then require the code comment to be moved to a less ideal place so I'm inclined to leave as is.

Reply:
The comment would be better in the body of the else if, as it would not need to repeat the "if removed":
```java
if (getHttpExchanges().isEmpty())
{
    tryRemoveIdleDestination();
}
else if (removed)
{
   // Process queued exchanges because waiting exchanges may now be able to progress. 
   process(true);
}